apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/PartitionOptions.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System.Threading;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// An immutable class representing partition options.
+/// </summary>
+/// <remarks>
+/// <para>
+/// The specified values of `PartitionSizeBytes` and `MaxPartitions`
+/// are utilized in the RPC call while generating the partition tokens.
+/// </para>
+/// <para>
+/// The specified flag for `DataBoostEnabled` will be stored in the request property
+/// of the <see cref="CommandPartition"/> object and is used when creating a request
+/// to read from the partition. If `DataBoostEnabled` is set to true, the request will be
+/// executed using Spanner-independent compute resources.
+/// </para>
+/// </remarks>
```

Comment:
nit: I think I would put all of this info on each of the individual properties. Maybe leaving a note here saying, "Some of the options are used when generating the partition tokens, while others are used when reading the partitions. Please refer to each of the options documentation for more information".